bin/cli.js

```diff
@@ -0,0 +1,271 @@
+      const subfiles = fs.readdirSync(file).map(f => path.join(file, f));
+      result.push(...loadFiles(subfiles));
+    } else if (common.fileExt(file) === 'js') {
+      result.push(path.relative(baseDir, file));
+    }
+  });
+  return result;
+};
+
+const setKarmaBrowsers = (browserConfig, ...browsers) => {
+  if (!browserConfig.plugins) browserConfig.plugins = [];
+  if (!browserConfig.browsers) browserConfig.browsers = [];
+
+  browsers.forEach(browser => {
+    const launcher = browserLaunchers[browser];
+    if (!launcher) {
+      throw new Error(
+        `Metatests library does not support such browser: ${browser}`
+      );
+    }
+    browserConfig.browsers.push(browser);
+    if (!browserConfig.plugins.includes(launcher)) {
+      browserConfig.plugins.push(launcher);
+    }
+  });
+};
+
+const setKarmaLogLevel = (browserConfig, logLevel = 'disable') => {
+  if (logLevel === 'quiet') logLevel = 'disable';
+  if (!browserConfig.webpackMiddleware) browserConfig.webpackMiddleware = {};
+  browserConfig.logLevel = karmaLogLevels[logLevel][0];
+  browserConfig.webpackMiddleware.stats = karmaLogLevels[logLevel][1];
+};
+
+const removeNodePackages = (browserConfig, ...packages) => {
+  if (!browserConfig.webpack) browserConfig.webpack = {};
+  if (!browserConfig.webpack.node) browserConfig.webpack.node = {};
+  packages.forEach(lib => {
+    browserConfig.webpack.node[lib] = 'empty';
+  });
+};
+
+const getReporter = logLevel => {
+  const browsers = [];
+  const reporter = function(base) {
+    base(this);
+    if (logLevel === 'quiet') {
+      this.onBrowserError = () => {};
+      this.onBrowserLog = () => {};
+      return;
+    }
+
+    this.onBrowserLog = (browser, log) => {
+      if (typeof log !== 'string') log = JSON.stringify(log, null, 2);
+      else log = log.slice(1, log.length - 1);
+      if (!browsers.includes(browser)) {
+        browsers.push(browser);
+        console.log(`\n${browser}:`);
+      }
+      console.log(log);
+    };
+
+    this.onBrowserError = (browser, error) => {
+      if (typeof error !== 'string') error = JSON.stringify(error, null, 2);
+      else error = error.slice(1, error.length - 1);
+      if (!browsers.includes(browser)) {
+        browsers.push(browser);
+        console.log(`\n${browser}:`);
+      }
+      console.error(error);
+    };
+  };
+
+  reporter.$inject = ['baseReporterDecorator'];
+  return reporter;
+};
+
+const getBrowserConfig = config => {
+  const browserConfig = {
+    preprocessors: {},
+    files: [],
+    plugins: [
+      'karma-webpack',
+      { 'reporter:meta': ['type', getReporter(config.browser.logLevel)] },
+    ],
+    reporters: [ 'meta' ],
+    basePath: baseDir,
+    port: config.browser.port,
+    autoWatch: false,
+    singleRun: true,
+    concurrency: 1,
+  };
+  const adapter = path.join(baseDir, 'build', 'adapter.js');
+  browserConfig.files.push(adapter);
+  browserConfig.preprocessors[adapter] = ['webpack'];
+  const nodePackages = [
```

Comment:
Could you extract this into file-variable at the top as this seems like a constant?